src/silo/query_engine/exec_node/table_scan.cpp

```diff
@@ -0,0 +1,188 @@
+      partition_reference[position_id] = SymbolType::symbolToChar(symbol);
+   }
+
+   std::vector<std::string> reconstructed_sequences(row_ids.cardinality(), partition_reference);
+
+   for (size_t position_id = 0; position_id < sequence_store.positions.size(); position_id++) {
+      const Position<SymbolType>& position = sequence_store.positions.at(position_id);
+      for (const auto symbol : SymbolType::SYMBOLS) {
+         if (position.isSymbolFlipped(symbol) || position.isSymbolDeleted(symbol)) {
+            continue;
+         }
+         for (size_t row_id : *position.getBitmap(symbol) & row_ids) {
+            reconstructed_sequences[row_id][position_id] = SymbolType::symbolToChar(symbol);
+         }
+      }
+   }
+
+   for (size_t row_id : row_ids) {
+      for (const size_t position_idx : sequence_store.missing_symbol_bitmaps.at(row_id)) {
+         reconstructed_sequences[row_id][position_idx] =
+            SymbolType::symbolToChar(SymbolType::SYMBOL_MISSING);
+      }
+   }
+   ARROW_RETURN_NOT_OK(output_array.AppendValues(reconstructed_sequences));
+   return arrow::Status::OK();
+}
+
+class ColumnEntryAppender {
+  public:
+   template <storage::column::Column Column>
+   arrow::Status operator()(
+      TableScan& table_scan_node,
+      const std::string& column_name,
+      const storage::TablePartition& table_partition,
+      const roaring::Roaring& row_ids
+   );
+};
+
+template <>
+arrow::Status ColumnEntryAppender::operator()<storage::column::SequenceColumnPartition<Nucleotide>>(
+   TableScan& table_scan_node,
+   const std::string& column_name,
+   const storage::TablePartition& table_partition,
+   const roaring::Roaring& row_ids
+) {
+   auto array =
+      table_scan_node
+         .getColumnTypeArrayBuilders<storage::column::SequenceColumnPartition<Nucleotide>>()
+         .at(column_name);
+   return appendSequences<Nucleotide>(
+      table_partition.columns.nuc_columns.at(column_name), row_ids, *array
+   );
+}
+
+template <>
+arrow::Status ColumnEntryAppender::operator()<storage::column::SequenceColumnPartition<AminoAcid>>(
+   TableScan& table_scan_node,
+   const std::string& column_name,
+   const storage::TablePartition& table_partition,
+   const roaring::Roaring& row_ids
+) {
+   auto array =
+      table_scan_node
+         .getColumnTypeArrayBuilders<storage::column::SequenceColumnPartition<AminoAcid>>()
+         .at(column_name);
+   return appendSequences<AminoAcid>(
+      table_partition.columns.aa_columns.at(column_name), row_ids, *array
+   );
+}
+
+template <storage::column::Column Column>
+arrow::Status ColumnEntryAppender::operator()(
+   TableScan& table_scan_node,
+   const std::string& column_name,
+   const storage::TablePartition& table_partition,
+   const roaring::Roaring& row_ids
+) {
+   auto array = table_scan_node.getColumnTypeArrayBuilders<Column>().at(column_name);
+   for (auto row_id : row_ids) {
+      using type = ArrowBuilderSelector<Column>::value_type;
```

Comment:
My IDE complains: "Dependent type requires using 'typename' keyword"
```suggestion
      using type = typename ArrowBuilderSelector<Column>::value_type;
```

Reply:
Interesting. But the compiler was happy?

Reply:
Yes, the compiler didn't complain apparently. Might also be that the IDE is wrong.